app/src/main/java/org/simple/clinic/patient/PatientSearchResult.kt

```diff
@@ -194,6 +194,23 @@ data class PatientSearchResult(
         ORDER BY priority ASC, phoneNumberPosition ASC
     """)
     fun searchByPhoneNumber2(phoneNumber: String, facilityId: UUID): List<PatientSearchResult>
+
+    @Query("""
+        SELECT 
+            searchResult.*, 
+            (
+                CASE
+                    WHEN P.assignedFacilityId = :facilityId THEN 0
+                    ELSE 1
+                END
+            ) AS priority,
+            INSTR(phoneNumber, :numericCriteria) phoneNumberPosition, 
+            INSTR(id_identifier, :numericCriteria) identifierPosition FROM PatientSearchResult searchResult
+        LEFT JOIN Patient P ON P.uuid = searchResult.uuid
+        WHERE P.deletedAt IS NULL AND phoneNumberPosition > 0 OR identifierPosition > 0
+        ORDER BY priority ASC, phoneNumberPosition ASC, identifierPosition ASC
+        """)
```

Comment:
This query can possible result in duplicate rows if either:
- A patient has multiple business identifiers that have a common sequence of numbers
- If a patient's phone number and their business ID have a common sequence of numbers

Reply:
I see that you added a `DISTINCT` clause to the SQL query. Was this meant to resolve this comment? This will not solve the issue because the `DISTINCT` clause operates on the entire row, what you might be looking for is the `GROUP BY` clause.

Please verify these changes using the instrumented tests when you make them.

Reply:
This comment also applies to all the queries you have added, not just this one specific one.

Reply:
I see there are checks in the test for this kind of data. However, the test is passing there because we are extracting the `UUID` instances from the search results and then converting it to a `Set`, which will remove duplicate `UUID` instances. If you convert it to a `List` instead, you'll see the duplicates showing up.

Reply:
I have added the `GROUPBY` clause. I have also added some duplicate values in the test to see if it's working properly. Firstly, I tested without `GROUPBY` and then with it. So, it is working correctly.